src/g_radio.c

```diff
@@ -48,113 +42,90 @@ void hradio_draw_update(t_gobj *client, t_glist *glist)
+        sys_vgui(".x%lx.c create rectangle %d %d %d %d -fill black -tags [list %lxIN%d %lxOBJ inlet]\n",
              canvas,
-             xx11b, yy11,
-             xx11b + iow, yy11 - IEMGUI_ZOOM(x) + ioh,
-             x, 0);
-    sys_vgui(".x%lx.c create text %d %d -text {%s} -anchor w -font {{%s} -%d %s} -fill #%06x -tags [list %lxLABEL label text]\n",
-             canvas, xx11b + x->x_gui.x_ldx * IEMGUI_ZOOM(x),
-             yy11 + x->x_gui.x_ldy * IEMGUI_ZOOM(x),
-             (strcmp(x->x_gui.x_lab->s_name, "empty") ? x->x_gui.x_lab->s_name : ""),
-             x->x_gui.x_font, x->x_gui.x_fontsize * IEMGUI_ZOOM(x), sys_fontweight,
-             x->x_gui.x_lcol, x);
+             xx11b, yy11b,
+             xx11b + iow, yy11b - IEMGUI_ZOOM(x) + ioh,
+            x, 0, x);
+
+    sys_vgui(".x%lx.c create text %d %d -text {%s} -anchor w -font {{%s} -%d %s} -fill #%06x -tags [list %lxLABEL %lxOBJ label text]\n",
+        canvas,
+        xx11b + x->x_gui.x_ldx * IEMGUI_ZOOM(x), yy11b + x->x_gui.x_ldy * IEMGUI_ZOOM(x),
+        (strcmp(x->x_gui.x_lab->s_name, "empty") ? x->x_gui.x_lab->s_name : ""),
+        x->x_gui.x_font, x->x_gui.x_fontsize * IEMGUI_ZOOM(x), sys_fontweight,
+        x->x_gui.x_lcol, x, x);
 }
 
-void hradio_draw_move(t_hradio *x, t_glist *glist)
+static void radio_draw_move(t_radio *x, t_glist *glist)
 {
-    int n = x->x_number, i, dx = x->x_gui.x_w, s4 = dx / 4;
-    int yy11 = text_ypix(&x->x_gui.x_obj, glist), yy12 = yy11 + dx;
-    int yy21 = yy11 + s4, yy22 = yy12 - s4;
-    int xx11b = text_xpix(&x->x_gui.x_obj, glist), xx11 = xx11b, xx21 = xx11b + s4;
-    int xx22 = xx11b + dx - s4;
-    int iow = IOWIDTH * IEMGUI_ZOOM(x), ioh = IEM_GUI_IOHEIGHT * IEMGUI_ZOOM(x);
     t_canvas *canvas = glist_getcanvas(glist);
-
-    xx11 = xx11b;
-    xx21 = xx11b + s4;
-    xx22 = xx11b + dx - s4;
-    for(i = 0; i < n; i++)
-    {
-        sys_vgui(".x%lx.c coords %lxBASE%d %d %d %d %d\n",
-                 canvas, x, i, xx11, yy11, xx11 + dx, yy12);
-        sys_vgui(".x%lx.c coords %lxBUT%d %d %d %d %d\n",
-                 canvas, x, i, xx21, yy21, xx22, yy22);
-        xx11 += dx;
-        xx21 += dx;
-        xx22 += dx;
-    }
-    sys_vgui(".x%lx.c coords %lxLABEL %d %d\n",
-             canvas, x, xx11b + x->x_gui.x_ldx * IEMGUI_ZOOM(x),
-             yy11 + x->x_gui.x_ldy * IEMGUI_ZOOM(x));
-    if(!x->x_gui.x_fsf.x_snd_able)
-        sys_vgui(".x%lx.c coords %lxOUT%d %d %d %d %d\n",
-             canvas, x, 0,
-             xx11b, yy12 + IEMGUI_ZOOM(x) - ioh,
-             xx11b + iow, yy12);
-    if(!x->x_gui.x_fsf.x_rcv_able)
-        sys_vgui(".x%lx.c coords %lxIN%d %d %d %d %d\n",
-             canvas, x, 0,
-             xx11b, yy11,
-             xx11b + iow, yy11 - IEMGUI_ZOOM(x) + ioh);
+    int dx = text_xpix(&x->x_gui.x_obj, glist) - x->x_gui.x_prevX;
+    int dy = text_ypix(&x->x_gui.x_obj, glist) - x->x_gui.x_prevY;
```

Comment:
I'm curious and this is somewhat OT, but when should we consider previous tcl/tk versions deprecated? obviously 8.4 is now, but it would be great to use the canvas "moveto" command introduced in 8.6 to avoid any unnecessary arithmetic in the audio thread (even if it is just 2 subtractions right now)..

Reply:
also, should the other iemguis also be updated to use "move" in this pr as well?

Reply:
if you manage to get Tcl/Tk to work on OSX-10.6/PowerPC, be my guest.

in the meantime, there are *many more* unnecessary calculations in the audio-thread that might be lifted to the GUI.

Reply:
.. my osx powerpc's hd went the way of the dodo ~7 yrs ago right after adobe discontinued flash support for it and then melted in a fire so I guess not

Reply:
on rethinking, i now believe that `moveto` is actually harmful: it assumes that the anchor point of the widget (which will eventually be positioned on the location declared by `moveto`) is the top-left corner of the bounding box.

that felt like a bad idea (conceptually), but i thought it would only be a theoretical problem, until i thought about *labels*.

Reply:
> also, should the other iemguis also be updated to use `move` in this pr as well?

yes.

Reply:
I guess that makes sense. hopefully in the future most click/edit handling will happen in the gui process anyways..